core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3049,6 +3049,13 @@ class KafkaApis(val requestChannel: RequestChannel,
       // Creating the shareFetchContext for Share Session Handling. if context creation fails, the request is failed directly here.
       shareFetchContext = sharePartitionManager.newContext(groupId, shareFetchData, forgottenTopics, newReqMetadata, isAcknowledgeDataPresent, request.context.connectionId)
     } catch {
+      case e: ShareSessionLimitReachedException =>
+        sharePartitionManager.createDelayedErrorFuture(shareFetchRequest.maxWait, e).whenComplete((_, exception) => {
+          if (exception != null) {
+            requestHelper.sendMaybeThrottle(request, shareFetchRequest.getErrorResponse(AbstractResponse.DEFAULT_THROTTLE_TIME, exception))
+          }
+        })
+        return
```

Comment:
Can't we use existing throttling to ask client not send next request as per max wait. I don't think we should go with the route of another scheduler thread to solve this issue. And I understand we can't block DataPlane thread either hence you created another thread. So my suggestion is to use throotling where client itself will not send the next fetch request. Can you check that once.

Reply:
In case we decide to go on broker to striclty wait then we should perhaps look for other solutions.

Reply:
Yes, I was expecting the request to be delayed using one of the existing mechanisms.